ActindoWorkFlow/UserInterface/01_WorkflowElements.md

```diff
@@ -0,0 +1,203 @@
+
+
+## Input and output ports
+
+![Input and output ports](/Assets/Screenshots/ActindoWorkFlow/Workflows/InputOutputPorts.png "[Input and output ports]")
+
+Every action can contain any number of input and output ports. They can be defined with a descriptive name, such as reporter, title or event, or be simply marked as *in* (input) and *out* (output) followed by a consecutive number.
+
+[comment]: <> (The plan is to provide all input and output ports with a name)
+
+Input ports are positioned to the left of the action and output ports are positioned to the right of the action. Both are represented by a square.
+
+Both input and output ports can be of two types:
+
+- **Mandatory**  
+  Mandatory ports must be linked to a place, which in turn must be linked to an action. Otherwise, the workflow stops and shows an error. Places linked to a mandatory port cannot be deleted or removed. When adding a new action, mandatory input/output places are already displayed.
+
+- **Optional**  
+ Optional ports are displayed with a ![Add](/Assets/Icons/Plus01.png "[Add]") (Add) button next to each of them. By clicking the ![Add](/Assets/Icons/Plus01.png "[Add]") (Add) button a new action can be added. For detailed information about adding new actions, see [Link](#to-be-completed).
+
+
+## Tokens
+
+![Tokens](/Assets/Screenshots/ActindoWorkFlow/Workflows/Tokens.png "[Tokens]")
+
+[comment]: <> (Token belongs in Processes)
+
+A token is a piece of data needed to complete an action, for example a document.
+
+The *Processes* menu entry displays a read-only view of any process taking place in real time and a list of all actions being executed in the selected process. For detailed information about tracking a workflow process, see [Link](#to-be-completed).
+
+A process is started with one token at the start point, which will be processed in each action executed and, if necessary, turned into several tokens. It is therefore possible that a place contains several tokens, for example, several delivery notes issued for different parts of a same order. Each token will then be processed separately, that means, that the subsequent actions will be executed once for each token available.
+
+The workflow process view displays the actions in different colours depending on their current status:
+
+  - Green: done
+  - Blue: active
+  - White: waiting
+  - Red: error
+
+[comment]: <> (Colour coding changing here too?)
+
+Next to the status of each action, a number is displayed. This number refers to the number of tokens processed in this action.  
+
+If a token is missing at some point, for example, if a document has not been received, the action cannot be executed and the process stops at the point where data are missing. In this case, a black square is displayed on the place specifying the number of tokens waiting to be processed.
```

Comment:
nicht komplett richtig -> all unprocessed tokens werden mit dem schwarzen rechteck angezeigt, da muss nicht unbedingt ein fehler vorliegen. Z.B. auch der Fall, wenn eine Aktion zwei Tokens aus unterschiedlichen places benötigt und das token noch nicht in einem der places vorhanden ist, weil eine vorherige Aktion noch nicht ausgeführt wurde -> vgl. Wissentransfer Part 1 ab Min. 41